packages/happychat-connection/src/use-happychat-available.ts

```diff
@@ -0,0 +1,21 @@
+import { useEffect, useState } from 'react';
+import buildConnection from './connection';
+import useHappychatAuth from './use-happychat-auth';
+
+export function useHappychatAvailable() {
+	const [ available, setIsAvailable ] = useState< boolean | undefined >( undefined );
+	const { data: dataAuth, isLoading: isLoadingAuth } = useHappychatAuth();
+
+	useEffect( () => {
+		if ( ! isLoadingAuth && dataAuth ) {
+			const connection = buildConnection( {
+				receiveAccept: ( receivedAvailability ) => setIsAvailable( receivedAvailability ),
```

Comment:
I think we should add `receiveUnauthorized` here and set `available` to false it's called.